src/LinAlg/SAM.C

```diff
@@ -976,23 +1006,24 @@ bool SAM::getNodalReactions (int inod, const Vector& rf, Vector& nrf) const
 }
 
 
-IntSet SAM::getEquations(char dofType, int dof) const
+IntSet SAM::getEquations (char dType, int dof) const
 {
   IntSet result;
-  for (int inod = 0; inod < nnod; inod++) {
-    char type = this->getNodeType(inod+1);
-    if (type == dofType || (dofType == 'D' && type == ' ')) {
-      if (dof > 0) {
-	int idof = madof[inod] + dof-1;
-	int ieq = idof < madof[inod+1] ? meqn[idof-1] : 0;
-	if (ieq > 0) result.insert(ieq);
-      }
-      else for (int idof = madof[inod]; idof < madof[inod+1]; idof++) {
-	int ieq = meqn[idof-1];
-	if (ieq > 0) result.insert(ieq);
-      }
+  auto&& getEquation = [this,dType,&result](int idof)
+  {
+    if ((dof_type.empty() ? 'D' : dof_type[idof]) == dType)
```

Comment:
dette feiler for petsc. vi har ingen dof_type, så når dType er 'P' går dette åt skogen. en 'fiks' som fungerer er å erstatte 'D' med dType.

Reply:
javel, men jeg trodde at hvis dof_type.empty() da evaluerer til true, så blir ikke dof_type[idof] eksekvert i denne konstruksjonen, eller tok jeg feil her..?

Reply:
nei, det har du helt rett i. problemet er at *dType* er 'P', mens vi alltid sammenligner med 'D' når dof_type er tom.

Reply:
Evt. slik da `if (dof_type.empty() || dof_type[idof] == dType)` i linje 1014. Prøver det jeg..